src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -399,6 +399,28 @@ class ConcatExpression : public detail::VariadicExpression {
 using EncodeForUriExpression =
     StringExpressionImpl<1, decltype(encodeForUriImpl)>;
 
+// TO_NUMERIC (int or double)
+auto InvalidArg = std::errc::invalid_argument;
+auto OutOfRange = std::errc::result_out_of_range;
+
+template <typename T>
+[[maybe_used]] auto toNumeric = [] (std::optional<std::string> input) {
+  if (!input.has_value()) {
+    return Id::makeUndefined();
+  } else {
+    T res{};
+    auto str = boost::lexical_cast<std::string>(*input);
+    auto conv = std::from_chars(str.data(), str.data() + str.size(), res);
+    if (conv.ec == InvalidArg || conv.ec == OutOfRange) {
```

Comment:
```suggestion
    if (conv.ec != std::error_code{} || conv.ptr != str.data() + str.size()) {
```

(The explicit enumeration of the possible errors is not needed, but you have to check for complete parsing, otherwise
something like `"42kartoffelsalat"` will be converted to `42` which we probably don't want.)

Reply:
Also add a test for this behavior.